src/java/org/apache/cassandra/service/accord/AccordCommand.java

```diff
@@ -652,59 +654,66 @@ private boolean canApplyWithCurrentScope(SafeCommandStore safeStore)
         return true;
     }
 
-    private Future<Void> applyWithCorrectScope(CommandStore unsafeStore)
+    private AsyncResult<Void> applyWithCorrectScope(CommandStore unsafeStore)
```

Comment:
Agreed, I applied the same modification in my branch. It’s not optimal, but it’s probably better than other approaches (another option is to offer a flatMap function in CommandStore, but it would likely be implemented the same way)